implementations/elixir/ockam/ockam_hub/lib/hub/service/token_lease_manager/cloud_service.ex

```diff
@@ -4,11 +4,50 @@ defmodule Ockam.TokenLeaseManager.CloudService do
   @type reason :: any()
   @type token_id :: String.t()
   @type options :: Keyword.t()
+  @type cloud_configuration :: Keyword.t()
+  @type state :: map()
+  @type creation_options :: map()
 
-  @callback configuration() :: Keyword.t()
-  @callback create(options) :: {:ok, lease} | {:error, reason}
-  @callback revoke(token_id) :: :ok | {:error, reason}
-  @callback renew(token_id) :: :ok | {:error, reason}
-  @callback get(token_id) :: {:ok, lease} | {:error, reason}
-  @callback get_all() :: {:ok, [lease]} | {:error, reason}
+  @callback do_init(options) :: {:ok, cloud_configuration} | {:error, reason}
+  @callback do_create(cloud_configuration, creation_options) :: {:ok, lease} | {:error, reason}
+  @callback do_revoke(cloud_configuration, token_id) :: :ok | {:error, reason}
+  @callback do_renew(cloud_configuration, token_id) :: :ok | {:error, reason}
+  @callback do_get(cloud_configuration, token_id) :: {:ok, lease} | :not_found | {:error, reason}
+  @callback do_get_all(cloud_configuration) :: {:ok, [lease]} | {:error, reason}
+  @callback do_get_address(cloud_configuration) :: {:ok, binary()} | {:error, reason}
+
+  defmacro __using__(_opts) do
```

Comment:
This macro might have been a module. It would be easier to debug. But that's my personal preference.

Reply:
It was a module, but I just wanted to make it a macro to make it closer to the storage one and to be prepared for future cloud providers that maybe need a long-term connection.

Reply:
In that case do we want to return potentially updated state from the `handle` functions? Or they're not supposed to change the state?